state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -327,29 +332,33 @@ impl DcaState {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, TypeInfo)]
-enum SwapRequestState {
-	Ccm {
-		gas_swap_state: GasSwapState,
-		ccm_deposit_metadata: CcmDepositMetadata,
-		output_address: ForeignChainAddress,
-		dca_state: DcaState,
-	},
-	Regular {
+struct CcmState {
+	gas_swap_state: GasSwapState,
+	ccm_deposit_metadata: CcmDepositMetadata,
+}
+
+#[allow(clippy::large_enum_variant)]
+#[derive(CloneNoBound, DebugNoBound, PartialEq, Eq, Encode, Decode, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+enum SwapRequestState<T: Config> {
+	UserSwap {
+		ccm: Option<CcmState>,
 		output_address: ForeignChainAddress,
 		dca_state: DcaState,
+		broker_fees: Beneficiaries<T::AccountId>,
 	},
 	NetworkFee,
 	IngressEgressFee,
```

Comment:
`SwapRequestState::NetworkFee`/`SwapRequestState::IngressEgressFee` are used in `process_swap_outcome` to determine the outcome. I think you may be referring to `SwapType` instead? That one is part of events, and I think it is useful to differentiate them there.

> We can refactor schedule_swap to take a list of FeeType instead of taking swap_type and broker_fees.

I suppose we can do this change regardless. I will give it a go.